### ISSUE TITLE
[Bug] queryParams cannot be updated from willTransition redirect for glob route

### ISSUE THREAD

**Issue by golampo** [emberjs/ember.js#19460]
### 🐞 Describe the Bug
A regression occurred in the 3.23.0 release such that: 
When issuing a new transition in the `willTransition` hook directly to a glob route, the queryParams will not be updated, and will remain their previous values.

### 🔬 Minimal Reproduction
**Ember Twiddle demo:**
Note that this is using Ember 3.18, so it works. If it was using 3.23.0 it would not work.
https://ember-twiddle.com/6169712a1330b377724f46cec767bed8?openFiles=routes.foo%5C.js%2C

**Quick contrived example:**

```
//templates/foo.js
<LinkTo @route=“foo” @model=“123” @query={{hash bar=“abc”}}>Click me</LinkTo> 
```

```
// routes/foo.js 
@action
willTransition() {
   this.router.replaceWith('foo', '987', { 
      queryParams: {  bar: 'xyz'  }
   });
  return true;
}
```

### 😕 Actual Behavior

Actual url for example: **/foo/987?bar=abc**

### 🤔 Expected Behavior

Expected url for example: **/foo/987?bar=xyz**

### 🌍 Environment

- Ember: 3.23.0

### ➕ Additional Context
This started in the 3.23.0 release. Works as expected in 3.22.2

**Comment by sly7-7** [comment]
Hello, there has been some fixes in this area introduced in the last 3.25.3 (and 3.24.3 maybe). It would be great if you can test against one of these version.

**Comment by golampo** [comment]
Closing this as I can't recreate the issue when doing a fresh install. Thank you!
